pyabc/sampler/redis_eps/sampler.py

```diff
@@ -636,3 +660,19 @@ def self_normalize_within_subpopulations(sample: Sample, n: int) -> Sample:
         particle.weight *= normalizations[particle.proposal_id]
 
     return sample
+
+
+def _log_active_set(
+    redis: StrictRedis, ana_id: str, t: int, id_results: List[Tuple],
+    batch_size: int,
+) -> None:
+    """Log the status of active simulations after the first n acceptances."""
+    accepted_ids = [id_result[0] for id_result in id_results]
+    active_set = get_active_set(redis=redis, ana_id=ana_id, t=t)
+    # remove entries that are already accepted (runtime conditions)
+    active_set = active_set.difference(accepted_ids)
+    earlier = {ix for ix in active_set if max(accepted_ids) > ix}
```

Comment:
So this "earlier" is the set we want to wait for, right? I dont see where you use that.

Reply:
This function here is only for logging. The collection is checked in

https://github.com/ICB-DCM/pyABC/pull/448/files#diff-2f41be0995fef5a262d8bf363d29590d9bec59065f8d219a2431ce40ce589344R263

There, it is checked whether it contains an entry with a smaller index than the maximum index of all so far accepted particles. Because it is those particles that could potentially, if accepted, replace that maximum index particle or others in the final accepted population.

Reply:
Right; saw it now. Should work imo :+1: